### PR TITLE
Fix put item

### PR DESCRIPTION
**Pull request by Elendan**

### CODE REVIEW

File: test/NosCore.Tests/HandlerTests/InventoryPacketControllerTests.cs [hunk 1]
```diff
@@ -179,8 +180,8 @@ public void Test_Put()
         [TestMethod]
         public void Test_PutBadPlace()
         {
-            _session.Character.PositionX = 3;
-            _session.Character.PositionY = 3;
+            _session.Character.PositionX = -13;
+            _session.Character.PositionY = -13;
```
Comment (0Lucifer0) [inline test/NosCore.Tests/HandlerTests/InventoryPacketControllerTests.cs]: please create your own test don't change mine ;) 3/3 is still a wall and shouldnt work

Comment (Elendan) [reply test/NosCore.Tests/HandlerTests/InventoryPacketControllerTests.cs]: Yeah but Putitem randomly selects some cells around the character

Comment (0Lucifer0) [reply test/NosCore.Tests/HandlerTests/InventoryPacketControllerTests.cs]: Yes this is why we use isblockedzone ;) and not iswalkable

File: test/NosCore.Tests/HandlerTests/InventoryPacketControllerTests.cs [hunk 2]
```diff
@@ -179,8 +182,8 @@ public void Test_Put()
         [TestMethod]
         public void Test_PutBadPlace()
         {
-            _session.Character.PositionX = 3;
-            _session.Character.PositionY = 3;
+            _session.Character.PositionX = -1;
+            _session.Character.PositionY = -1;
```
Comment (0Lucifer0) [inline test/NosCore.Tests/HandlerTests/InventoryPacketControllerTests.cs]: copy this test and do a 2/2

Comment (0Lucifer0) [reply test/NosCore.Tests/HandlerTests/InventoryPacketControllerTests.cs]: call the -1 -1 PutOutBoundaries

File: src/NosCore.GameObject/ComponentEntities/Extensions/VisualEntityExtension.cs [hunk 3]
```diff
@@ -196,7 +196,6 @@ public static InPacket GenerateIn(this ICountableEntity visualEntity)
                 VNum = visualEntity.VNum == 0 ? string.Empty : visualEntity.VNum.ToString(),
                 PositionX = visualEntity.PositionX,
                 PositionY = visualEntity.PositionY,
-                Direction = visualEntity.Direction,
```
Comment (0Lucifer0) [inline src/NosCore.GameObject/ComponentEntities/Extensions/VisualEntityExtension.cs]: ?

File: src/NosCore.GameObject/ComponentEntities/Extensions/ItemInstanceExtension.cs [hunk 4]
```diff
@@ -27,7 +27,7 @@ namespace NosCore.GameObject.ComponentEntities.Extensions
 {
     public static class IItemInstanceExtension
     {
-        public static IvnPacket GeneratePocketChange(this IItemInstance itemInstance, PocketType type, short slot)
+        public static IvnPacket GeneratePocketChange(this IItemInstance itemInstance, PocketType type, short slot, bool deleteEquip)
```
Comment (erwan-joly) [inline src/NosCore.GameObject/ComponentEntities/Extensions/ItemInstanceExtension.cs]: useless it's already handle if your iteminstance is null

Comment (erwan-joly) [reply src/NosCore.GameObject/ComponentEntities/Extensions/ItemInstanceExtension.cs]: juste use ((iteminstance)null).GeneratePocketChange

File: src/NosCore.GameObject/ComponentEntities/Extensions/ItemInstanceExtension.cs [hunk 5]
```diff
@@ -56,7 +56,7 @@ public static IvnPacket GeneratePocketChange(this IItemInstance itemInstance, Po
                     new IvnSubPacket
                     {
                         Slot = slot,
-                        VNum = itemInstance.ItemVNum,
+                        VNum = (short)(itemInstance.Type == PocketType.Equipment && deleteEquip ? -1 :  itemInstance.ItemVNum),
```
Comment (erwan-joly) [inline src/NosCore.GameObject/ComponentEntities/Extensions/ItemInstanceExtension.cs]: useless change

File: src/NosCore.GameObject/Services/Inventory/InventoryService.cs [hunk 6]
```diff
@@ -239,7 +239,7 @@ public IItemInstance MoveInPocket(short sourceSlot, PocketType sourceType, Pocke
             }
 
             var sourceInstance = LoadBySlotAndType<IItemInstance>(sourceSlot, sourceType);
-            if (!(sourceInstance is WearableInstance || sourceInstance is SpecialistInstance))
+            if (!(sourceInstance is WearableInstance || sourceInstance is SpecialistInstance && (targetType != PocketType.Equipment || targetType != PocketType.Specialist)))
```
Comment (erwan-joly) [inline src/NosCore.GameObject/Services/Inventory/InventoryService.cs]: useless check.

File: src/NosCore.Controllers/InventoryPacketController.cs [hunk 7]
```diff
@@ -63,8 +63,8 @@ public void MoveEquipment(MvePacket mvePacket)
 
             var inv = Session.Character.Inventory.MoveInPocket(mvePacket.Slot, mvePacket.InventoryType,
                 mvePacket.DestinationInventoryType, mvePacket.DestinationSlot, false);
-            Session.SendPacket(inv.GeneratePocketChange(mvePacket.DestinationInventoryType, mvePacket.DestinationSlot));
-            Session.SendPacket(((IItemInstance) null).GeneratePocketChange(mvePacket.InventoryType, mvePacket.Slot));
+            Session.SendPacket(inv.GeneratePocketChange(mvePacket.DestinationInventoryType, mvePacket.DestinationSlot, false));
+            Session.SendPacket(((IItemInstance) null).GeneratePocketChange(mvePacket.InventoryType, mvePacket.Slot, false));
```
Comment (erwan-joly) [inline src/NosCore.Controllers/InventoryPacketController.cs]: useless and also wrong. This line is a delete

File: src/NosCore.GameObject/Services/Inventory/InventoryService.cs (hunk 6 above)
Comment (Elendan) [reply src/NosCore.GameObject/Services/Inventory/InventoryService.cs]: Nope, specialist cards can be moved from Eq to Specialist card inventory

File: src/NosCore.Controllers/InventoryPacketController.cs [hunk 8]
```diff
@@ -263,7 +263,14 @@ public void PutItem(PutPacket putPacket)
                                 Session.Character.Inventory.DeleteFromTypeAndSlot(invitem.Type, invitem.Slot);
                             }
 
-                            Session.SendPacket(invitem.GeneratePocketChange(invitem.Type, invitem.Slot));
+                            if (invitem.Type == PocketType.Equipment)
+                            {
+                                Session.SendPacket(((IItemInstance)null).GeneratePocketChange(invitem.Type, invitem.Slot));
```
Comment (erwan-joly) [inline src/NosCore.Controllers/InventoryPacketController.cs]: not sure to understand the goal of this

File: src/NosCore.GameObject/Services/Inventory/InventoryService.cs (hunk 6 above)
Comment (erwan-joly) [reply src/NosCore.GameObject/Services/Inventory/InventoryService.cs]: yes but it's handle after line 251 252 here we are just checking only specialist card or wearable are going to the next part

File: src/NosCore.GameObject/Services/Inventory/InventoryService.cs [hunk 9]
```diff
@@ -248,8 +248,8 @@ public IItemInstance MoveInPocket(short sourceSlot, PocketType sourceType, Pocke
 
             switch (sourceInstance?.Item.ItemType)
             {
-                case ItemType.Fashion when targetType != PocketType.Main && targetType != PocketType.Costume:
-                case ItemType.Specialist when targetType != PocketType.Main && targetType != PocketType.Specialist:
+                case ItemType.Fashion when targetType != PocketType.Equipment && targetType != PocketType.Costume:
```
Comment (erwan-joly) [inline src/NosCore.GameObject/Services/Inventory/InventoryService.cs]: in fact this code is useless we should move it to a specific place just over we should handle 
not ItemType.Fashion || Sp when targetType == PocketType.Sp || targetType == PocketType.Costume

File: src/NosCore.GameObject/Services/Inventory/InventoryService.cs (hunk 6 above)
Comment (erwan-joly) [reply src/NosCore.GameObject/Services/Inventory/InventoryService.cs]: NVM those line are good but difficult to read you may need to rework this and delete the line 251 252

File: src/NosCore.GameObject/Services/Inventory/InventoryService.cs [hunk 10]
```diff
@@ -248,8 +248,8 @@ public IItemInstance MoveInPocket(short sourceSlot, PocketType sourceType, Pocke
 
             switch (sourceInstance?.Item.ItemType)
```
Comment (erwan-joly) [inline src/NosCore.GameObject/Services/Inventory/InventoryService.cs]: it wont need a switch if we only are keeping the default

**Pull request merged**